core/src/main/java/io/questdb/cairo/vm/api/MemoryCARW.java

```diff
@@ -0,0 +1,247 @@
+                value.getLong3());
+    }
+
+    default void putLong256(CharSequence hexString) {
+        final int len;
+        if (hexString == null || (len = hexString.length()) == 0) {
+            putLong256Null();
+        } else {
+            putLong256(hexString, 2, len);
+        }
+    }
+
+    default long putNullBin() {
+        putLong(TableUtils.NULL_LEN);
+        return getAppendOffset();
+    }
+
+    default long putNullStr() {
+        putInt(TableUtils.NULL_LEN);
+        return getAppendOffset();
+    }
+
+    default void putShort(short value) {
+        Unsafe.getUnsafe().putShort(appendAddressFor(Short.BYTES), value);
+    }
+
+    default long putStr(CharSequence value) {
+        return value != null ? putStrUnsafe(value, 0, value.length()) : putNullStr();
+    }
+
+    default long putStr(char value) {
+        if (value != 0) {
+            long addr = appendAddressFor(Integer.BYTES + Character.BYTES);
+            Unsafe.getUnsafe().putInt(addr, 1);
+            Unsafe.getUnsafe().putChar(addr + Integer.BYTES, value);
+            return getAppendOffset();
+        }
+        return putNullStr();
+    }
+
+    default long putStr(CharSequence value, int pos, int len) {
+        if (value != null) {
+            return putStrUnsafe(value, pos, len);
+        }
+        return putNullStr();
+    }
+
+    default void putBool(long offset, boolean value) {
+        putByte(offset, (byte) (value ? 1 : 0));
+    }
+
+    default void putByte(long offset, byte value) {
+        Unsafe.getUnsafe().putByte(appendAddressFor(offset, Byte.BYTES), value);
+    }
+
+    default void putChar(long offset, char value) {
+        Unsafe.getUnsafe().putChar(appendAddressFor(offset, Character.BYTES), value);
+    }
+
+    default void putDouble(long offset, double value) {
+        Unsafe.getUnsafe().putDouble(appendAddressFor(offset, Double.BYTES), value);
+    }
+
+    default void putFloat(long offset, float value) {
+        Unsafe.getUnsafe().putFloat(appendAddressFor(offset, Float.BYTES), value);
+    }
+
+    default void putInt(long offset, int value) {
+        Unsafe.getUnsafe().putInt(appendAddressFor(offset, Integer.BYTES), value);
+    }
+
+    default void putLong(long offset, long value) {
+        Unsafe.getUnsafe().putLong(appendAddressFor(offset, Long.BYTES), value);
+    }
+
+    default void putLong256(long offset, Long256 value) {
+        putLong256(
+                offset,
+                value.getLong0(),
+                value.getLong1(),
+                value.getLong2(),
+                value.getLong3());
+    }
+
+    default void putLong256(long offset, long l0, long l1, long l2, long l3) {
+        final long addr = appendAddressFor(offset, Long256.BYTES);
+        Unsafe.getUnsafe().putLong(addr, l0);
+        Unsafe.getUnsafe().putLong(addr + Long.BYTES, l1);
+        Unsafe.getUnsafe().putLong(addr + Long.BYTES * 2, l2);
+        Unsafe.getUnsafe().putLong(addr + Long.BYTES * 3, l3);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).